lib/crossbar.ex

```diff
@@ -91,19 +91,9 @@ defmodule Crossbar do
   The default value of `8080` for websocket and `9000` for raw_socket can be
   overrode using the environment variables `CROSSBAR_PORT_WS` and `CROSSBAR_PORT_RS`.
   """
-  @spec get_port(:websocket | :raw_socket) :: :inet.port
-  def get_port(:websocket) do
-    case System.get_env("CROSSBAR_PORT_WS") do
-      nil -> 8080
-      port when is_binary(port) -> String.to_integer(port)
-    end
-  end
-  def get_port(:raw_socket) do
-    case System.get_env("CROSSBAR_PORT_RS") do
-      nil -> 9000
-      port when is_binary(port) -> String.to_integer(port)
-    end
-  end
+  @spec get_port(String.t) :: :inet.port
```

Comment:
I think here you can define a type for transport and then use it, something like:

```elixir
@type transport_type :: "websocket" | "raw_socket"
```